x-pack/legacy/plugins/code/server/utils/detect_language.ts

```diff
@@ -4,15 +4,54 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import path from 'path';
+import fs from 'fs';
 
-import * as extensions from './extensions.json';
+function matchAll(s: string, re: RegExp): string[][] {
+  let m;
+  const result = [];
+  while ((m = re.exec(s)) !== null) {
+    // This is necessary to avoid infinite loops with zero-width matches
+    if (m.index === re.lastIndex) {
+      re.lastIndex++;
+    }
+    result.push(m);
+  }
+  return result;
+}
+
+const entryFilePath = require.resolve('monaco-editor/esm/vs/basic-languages/monaco.contribution');
+let languageMap: { [key: string]: string } = {};
+if (entryFilePath) {
+  const entryFileContent = fs.readFileSync(entryFilePath, 'utf8');
+  const importRegex = /import '(.*\.contribution\.js)'/gm;
+  const dir = path.dirname(entryFilePath);
+  const regex = /id:\s*'(.*)',\s*extensions:\s*\[(.+)\]/gm;
 
-const languageMap: { [key: string]: string } = extensions;
+  for (const m of matchAll(entryFileContent, importRegex)) {
+    const contributionFile = path.join(dir, m[1]);
+    if (fs.existsSync(contributionFile)) {
+      const contributionContent = fs.readFileSync(contributionFile, 'utf8');
+      for (const mm of matchAll(contributionContent, regex)) {
+        const langId = mm[1];
+        const extensions = mm[2];
+        for (let ext of extensions.split(',')) {
+          ext = ext.trim().slice(1, -1);
+          languageMap[ext] = langId;
+        }
+      }
+    }
+  }
+} else {
+  // fallback to extension.json
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const extensions = require('./extensions.json');
 
-// patch the lib
-languageMap['.ts'] = 'typescript';
-languageMap['.tsx'] = 'typescript';
+  languageMap = extensions;
```

Comment:
do we still need this?

maybe just let test fail if the first branch doesn't get executed